annotation-statistics/src/main/java/eu/europeana/annotation/statistics/service/AnnotationStatisticsService.java

```diff
@@ -0,0 +1,93 @@
+import eu.europeana.annotation.statistics.model.AnnotationStatistics;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsClient;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsUser;
+import eu.europeana.annotation.statistics.vocabulary.AnnotationStatisticsConstants;
+
+public class AnnotationStatisticsService {
+
+	@Resource
+	SolrAnnotationService solrService;
+
+    public void getAnnotationsStatistics(AnnotationMetric annoMetric) throws AnnotationServiceException {
+    	annoMetric.setTimestamp(new Date());  
+    	//getting the annotations statistics for the clients
+    	QueryResponse annoFacetStatsJson = solrService.getAnnotationStatisticsJsonNestedFacets(SolrAnnotationConstants.GENERATOR_URI);
+    	extractAnnotationStatisticsClientUserFromJsonNestedFacet(annoFacetStatsJson, annoMetric, SolrAnnotationConstants.GENERATOR_URI, AnnotationStatisticsConstants.CLIENT);   	
+    	//getting the annotations statistics for the users
+    	annoFacetStatsJson = solrService.getAnnotationStatisticsJsonNestedFacets(SolrAnnotationConstants.CREATOR_URI);
+    	extractAnnotationStatisticsClientUserFromJsonNestedFacet(annoFacetStatsJson, annoMetric, SolrAnnotationConstants.CREATOR_URI, AnnotationStatisticsConstants.USER);   	
+    	//getting the overall annotations statistics
+    	annoFacetStatsJson = solrService.getAnnotationStatisticsJsonFacets(); 
+    	extractAnnotationStatisticsOverallFromJsonFacet(annoFacetStatsJson, annoMetric);
+    }
+    
+    private void extractAnnotationStatisticsClientUserFromJsonNestedFacet (QueryResponse annoFacetStats, AnnotationMetric annoMetric, String facetField, String target) {
+    	List<AnnotationStatistics> annotationStatistics = new ArrayList<AnnotationStatistics>();
+	    annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets();
+    	for (BucketJsonFacet mainBucket : annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets()) {
+    		AnnotationStatistics annoStats = null;
+    		if(target.compareToIgnoreCase(AnnotationStatisticsConstants.CLIENT)==0) {
+    			annoStats = new AnnotationStatisticsClient();
+    			((AnnotationStatisticsClient)annoStats).setClient(mainBucket.getVal().toString());
+    		}
+    		else if(target.compareToIgnoreCase(AnnotationStatisticsConstants.USER)==0) {
+    			annoStats = new AnnotationStatisticsUser();
+    			((AnnotationStatisticsUser)annoStats).setUser(mainBucket.getVal().toString());
+    		}
+    		
+    		if (annoStats==null) continue;
+		    	
+    		setAnnotationStatisticsFromJsonBuckets(annoStats, mainBucket.getBucketBasedFacets(SolrAnnotationConstants.SCENARIO).getBuckets());
+        	annotationStatistics.add(annoStats);
+    	}
+	    
+		if(target.compareToIgnoreCase(AnnotationStatisticsConstants.CLIENT)==0) {
+			annoMetric.setAnnotationStatisticsClients(annotationStatistics);
+		}
+		else if(target.compareToIgnoreCase(AnnotationStatisticsConstants.USER)==0) {
```

Comment:
please switch the position of the constant and use equalsIgnoreCase

Reply:
This code is not needed any more.